app/models/grda_warehouse/hmis_client.rb

```diff
@@ -26,33 +26,15 @@ class GrdaWarehouse::HmisClient < GrdaWarehouseBase
     where.not(id: consent_active.select(:id))
   end
 
-  NominatimApiPaused = Class.new(StandardError)
-
   def address_lat_lon
-    address = last_permanent_zip
-    return unless address.present?
-
-    begin
-      result = Rails.cache.fetch(['Nominatim', address.to_s], expires_in: 6.weeks) do
-        raise(NominatimApiPaused, 'Nominatim Paused') if Rails.cache.read(['Nominatim', 'API PAUSE'])
-
-        sleep(0.75)
-        begin
-          Nominatim.search(address).country_codes('us').first
-        rescue Faraday::ConnectionFailed
-          # we've probably been banned, let the API cool off
-          Rails.cache.write(['Nominatim', 'API PAUSE'], true, expires_in: 1.hours)
-          raise(NominatimApiPaused, 'Nominatim Paused')
-        end
-      end
-      return { address: address, lat: result.lat, lon: result.lon, boundingbox: result.boundingbox } if result.present?
-    rescue NominatimApiPaused
-      # Ignore errors if we are paused
-    rescue StandardError
-      setup_notifier('NominatimWarning')
-      @notifier.ping("Error contacting the OSM Nominatim API. Looking address for enrollment id: #{id}") if @send_notifications
-    end
-    return nil
+    return unless last_permanent_zip.present?
+
+    lat, lon, bound = Place.lookup_lat_lon(postalcode: last_permanent_zip)
+    return { address: address, lat: lat, lon: lon, boundingbox: bound } if bound.present?
+  rescue StandardError
+    setup_notifier('NominatimWarning')
+    @notifier.ping("Error contacting the OSM Nominatim API. Looking address for HmisClient id: #{id}") if @send_notifications
+    nil
```

Comment:
This and the similar one in enrollment.rb should be moved to place.rb#nominatim_lookup no?

Reply:
I had left it here, because there is some information about what we were using Nominatim for: `Looking address for HmisClient id: #{id}` -- if that isn't really important -- we can push it down.

Reply:
Let's push it down.  Thanks!